src/test/java/codeu/controller/ChatServletTest.java

```diff
@@ -207,7 +207,7 @@ public void testDoPost_CleansHtmlContent() throws IOException, ServletException
     ArgumentCaptor<Message> messageArgumentCaptor = ArgumentCaptor.forClass(Message.class);
     Mockito.verify(mockMessageStore).addMessage(messageArgumentCaptor.capture());
     Assert.assertEquals(
-        "Contains html and  content.", messageArgumentCaptor.getValue().getContent());
+        "Contains <b>html</b> and  content.", messageArgumentCaptor.getValue().getContent());
 
     Mockito.verify(mockResponse).sendRedirect("/chat/test_conversation");
   }
```

Comment:
Maybe add another test that verifies that unsafe HTML tags (like `<script>`) are still filtered out

Reply:
The original message was "Contains <b>html</b> and <script>JavaScript</script> content." and was filtered to 
"Contains <b>html</b> and  content." Should I try another one besides <script>?

Reply:
I think your test should actually be okay as for checking script like you said, on my first review of the code I also missed the fact that the original message has the <script> tag inside it a few lines before: Mockito.when(mockRequest.getParameter("message"))
        .thenReturn("Contains <b>html</b> and <script>JavaScript</script> content."); 
We can wait for Vasu to double check if this test is okay though!

My suggestion would be exactly what you just said though, to maybe add a couple more tags to check for in the message (both safe and unsafe). Another good unsafe tag to check for being filtered out would probably be the img tag since the docs say that Whitelist.basic() disallows images. https://jsoup.org/apidocs/org/jsoup/safety/Whitelist.html#basic--

Reply:
Yes, `<img>` tags seem like a good candidate to test for. Also from the documentation linked, it looks like links (`<a>` tags) are allowed, we should check that the URLs (relative and absolute) are handled properly